engine/src/types.rs

```diff
@@ -179,6 +179,17 @@ impl<'a> From<&'a RhsValue> for LhsValue<'a> {
     }
 }
 
+impl<'a> From<&'a LhsValue<'_>> for LhsValue<'a> {
```

Comment:
This might be confusing to use properly. I'd suggest changing it to a custom method instead. For example, it could be called `as_ref` to align with [`Option::as_ref`](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_ref) which does something similar (turns reference to an `Option` to an `Option` with an internal reference, just like you turn reference to `LhsValue` to a `LhsValue` with an internal reference).

Reply:
Also, I believe same method could be reused in `get_field_value_unchecked` too?

Reply:
Good point for `get_field_value_unchecked`, i'll fix all of this asap